src/box/alter.cc

```diff
@@ -1156,6 +1157,9 @@ alter_space_do(struct txn_stmt *stmt, struct alter_space *alter)
 	space_fill_index_map(alter->old_space);
 	space_fill_index_map(alter->new_space);
 
+	if (space_is_memtx(alter->new_space))
```

Comment:
I think that is a bad dependency on memtx. We should avoid that.
Perhaps it's better to update to update index vtabs in  the beginning of `memtx_space_replace_all_keys`. There must a very simple condition to check whether index vtabs must be updated.

Reply:
I implement `memtx_space_swap_index` function instead of using it's genetic version and update indexes vtab in this function.